extensions/amp-iframely/0.1/amp-iframely.js

```diff
@@ -0,0 +1,417 @@
+ * @param {document.window} contentWindow - source of a received message
+ * @return {iframe} the matched iFrame
+ * */
+function findIframeByContentWindow(iframes, contentWindow) {
+  let selectedIframe = false;
+  for (let i = 0; i < iframes.length && !selectedIframe; i++) {
+    const iframe = iframes[i];
+    if (iframe.contentWindow === contentWindow) {
+      selectedIframe = iframe;
+    }
+  }
+  return selectedIframe;
+}
+
+/**
+ * Implementation of the amp-iframely component.
+ *
+ * See {@link ../amp-iframely.md} for the spec.
+ */
+export class AmpIframely extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /**
+     * Address of Iframely media (3rd party rich media or a summary card for the URL).
+     * @private {string}
+     */
+    this.src_ = null;
+
+    /**
+     * The main iFrame element that loads Iframely media.
+     * @private {?Element}
+     */
+    this.iframe_ = null;
+
+    /**
+     * ID of Iframely content, if available
+     * @private {?string}
+     */
+    this.widgetId_ = null;
+
+    /**
+     * Domain of Iframely CDN
+     * @private {?string}
+     */
+    this.base_ = null;
+
+    /**
+     * Alternatively, identify with Iframely CDN via `url` and `key` hash params
+     * @private {?string}
+     */
+    this.url_ = null;
+    /** @private {?string} */
+    this.key_ = null;
+
+    /**
+     * other data- options that will be passed into iFrame src
+     * @private {?Object}
+     * */
+    this.options_ = null;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    /** Populating parameters once */
+    this.widgetId_ = this.element.getAttribute('data-id');
+    this.url_ = this.element.getAttribute('data-url');
+    this.key_ = this.element.getAttribute('data-key');
+    this.options_ = this.parseOptions_();
+    let domain = 'cdn.iframe.ly';
+    const requestedDomain = this.element.getAttribute('data-domain');
+    if (requestedDomain && this.isValidDomain_(requestedDomain)) {
+      domain = requestedDomain;
+    }
+    this.base_ = `https://${domain}/`;
+    this.parseAttributes_();
+    setStyle(this.element, 'box-sizing', 'border-box');
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    if (this.preconnect && this.preconnect.url_) {
+      this.preconnect.url(this.base_, opt_onLayout);
+    }
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    /**
+     * Building placeholder for responsive layout without resizing
+     * or when requested via `data-img` attribute
+     */
+    const layout = this.getLayout();
+    const resizable = this.element.getAttribute('resizable') !== null;
+    const isImg = this.element.getAttribute('data-img') !== null;
+
+    if (isImg || (layout === 'responsive' && !resizable)) {
```

Comment:
Consider `hasAttribute` and inlining.
```suggestion
    if (
      this.element.hasAttribute("data-img") ||
      (layout === "responsive" && !this.element.hasAttribute("resizable"))
    ) {
```